HelpSource/Classes/Pclump.schelp

```diff
@@ -0,0 +1,76 @@
+TITLE:: Pclump
+summary:: Clumps the pattern into arrays of size n.
+categories:: Streams-Patterns-Events>Patterns>Repetition
+
+DESCRIPTION::
+Takes the source pattern and clumps it into arrays of size n, with the final element being the size of the remainder.
+
+E.g. If the source pattern has 5 elements and you choose a clump size of 2, the new pattern will return two arrays of size 2, and a final array of size 1.
+
+CLASSMETHODS::
+
+METHOD:: new
+
+ARGUMENT:: pattern
+pattern to be filtered
+
+ARGUMENT:: n
+a pattern that returns an integer. The returned value will determine the size of the next clump.
+
+EXAMPLES::
+
+code::
```

Comment:
IMHO one example a la

```
x = Pclump(2, Pseq([1,2,3], 2)).asStream

// repeat this
x.next

// -> [ 1, 2 ]
// -> [ 3, 1 ]
// -> [ 2, 3 ]
// -> nil
```

should be sufficient.

Reply:
Maybe. I wanted there to at least be an example that made it clear that if there's a remainder the final element is the remainder. When writing requirements for my day job - I find that examples that highlight edge cases have been really useful in preventing bugs further down the line. I'll look at this again.